src/app.ts

```diff
@@ -30,62 +30,98 @@ function getCookieString(req: express.Request) {
+		res.send({
+			error: 'Service incorrectly configured. No salt provided',
+		});
+		return;
+	}
+
+	const profile = config?.profile ?? { width: 800 };
+
+	try {
+		const signedUrl = format(url, salt, profile);
+		res.send({ signedUrl });
+	} catch (ex: unknown) {
+		res.status(500).send({
+			error: 'Error signing url',
+			ex: ex,
+		});
+	}
+}
+
+function withPandaAuth(
+	getPanda: () => PanDomainAuthentication,
+	req: express.Request,
+	res: express.Response,
+	onSuccess: () => unknown,
+) {
+	const panda = getPanda();
+	panda
+		.verify(getCookieString(req))
+		.then((panAuthResult) => {
+			if (panAuthResult.status === 'Authorised') {
+				onSuccess();
+			} else {
+				res.status(403).send({
+					error: 'Not authorised by pan-domain login',
+				});
+			}
+		})
+		.catch((ex: unknown) => {
+			res.status(500).send({
+				error: 'Pan domain auth error',
+				ex: ex,
+			});
+		});
+}
+
 export function buildApp(
 	getPanda: () => PanDomainAuthentication = getPanDomainAuth,
 ): Express {
 	const app = express();
 	app.use(jsonBodyParser());
 
-	app.all(
+	app.post(
 		'/signed-image-url',
-		(req: express.Request, res: express.Response) => {
-			// Allow setting of image URL via body or query parameter
-			const body = req.body as SignedImageUrlBody | undefined;
-			const url = body?.url ?? req.query['url'];
-			if (!url || typeof url != 'string' || url.length <= 0) {
-				res.status(400);
-				res.send({ error: 'No URL provided' });
-				return;
-			}
-
-			const salt = process.env.SALT;
-			if (!salt) {
-				res.status(500);
-				res.send({
-					error: 'Service incorrectly configured. No salt provided',
-				});
-				return;
-			}
-
-			const profile = body?.profile ?? { width: 800 };
+		(req: express.Request, res: express.Response) =>
+			withPandaAuth(getPanda, req, res, () => {
+				handleImageSigning(
+					req.body as SignedImageUrlConfig | undefined,
+					getPanda,
+					req,
+					res,
+				);
+			}),
+	);
 
-			const panda = getPanda();
-			panda
-				.verify(getCookieString(req))
-				.then((panAuthResult) => {
-					if (panAuthResult.status === 'Authorised') {
-						try {
-							const signedUrl = format(url, salt, profile);
-							res.send({ signedUrl });
-						} catch (ex: unknown) {
-							res.status(500).send({
-								error: 'Error signing url',
-								ex: ex,
-							});
-						}
-					} else {
-						res.status(403).send({
-							error: 'Not authorised by pan-domain login',
-						});
-					}
-				})
-				.catch((ex: unknown) => {
-					res.status(500).send({
-						error: 'Pan domain auth error',
-						ex: ex,
-					});
-				});
-		},
+	app.get(
+		'/signed-image-url',
+		(req: express.Request, res: express.Response) =>
+			withPandaAuth(getPanda, req, res, () => {
+				const config = { url: req.query.url };
+				handleImageSigning(
+					config as SignedImageUrlConfig,
+					getPanda,
+					req,
+					res,
+				);
+			}),
 	);
 
 	app.get('/userdetails', (req: express.Request, res: express.Response) => {
```

Comment:
@michaelclapham I've not touched `/userdetails` in this refactoring but I think it'd be straightforward to also use `withPandaAuth` in this case. I noticed that there are some slight differences between the auth wiring here vs `/signed-image-url`, for example we don't call `panda.stop()` for `/signed-image-url` but we do here. Is that difference deliberate do you know? Or should we be calling stop in all cases?

Reply:
We should be calling stop in all cases.

Reply:
Thanks 👍  I've updated the change to do this - the `withPandaAuth` helper now calls `stop()` in a `finally` block, and the `/userdetails` endpoint has been updated to use `withPandaAuth`.